src/main/java/io/github/ImpactDevelopment/installer/setting/settings/OptiFineFileSetting.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * This file is part of Impact Installer.
+ *
+ * Copyright (C) 2019  ImpactDevelopment and contributors
+ *
+ * See the CONTRIBUTORS.md file for a list of copyright holders
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package io.github.ImpactDevelopment.installer.setting.settings;
+
+import io.github.ImpactDevelopment.installer.setting.InstallationConfig;
+import io.github.ImpactDevelopment.installer.setting.Setting;
+import io.github.ImpactDevelopment.installer.utils.OperatingSystem;
+
+import java.nio.file.Path;
+
+public enum OptiFineFileSetting implements Setting<Path> {
+    INSTANCE;
+
+    @Override
+    public Path getDefaultValue(InstallationConfig config) {
+            return OperatingSystem.getDownloads();
+    }
+
+    @Override
+    public boolean validSetting(InstallationConfig config, Path value) {
+//        return value.getFileName().toString().toLowerCase().endsWith(".jar");
```

Comment:
This seems like something good to include?

Reply:
It causes issues when the jar hasn't been selected yet, because the default value is `~/Downloads` not `anything.jar`